File: app/rotas/modulos.py
"""/modulos - o mapa do sistema - e /inicio, o despachante da porta da frente."""

from __future__ import annotations

from fastapi import APIRouter, Request
from fastapi.responses import RedirectResponse

from app import modulos
from app.dependencias import UsuarioDep
from app.web import pagina

rotas = APIRouter(tags=["modulos"])


@rotas.get("/modulos")
def listar(request: Request, usuario: UsuarioDep):
    # sem exigir permissao: e a tela que explica onde ficam as coisas, e cada
    # item ja se esconde sozinho de quem nao pode ve-lo
    return pagina(request, "paginas/modulos.html", usuario=usuario)


@rotas.get("/inicio")
def inicio(usuario: UsuarioDep):
    """Onde o sistema comeca para QUEM entrou — nao e tela, e desvio.

    `/` e o painel de Processos SEI e exige `processo.ver`. Enquanto todo perfil
    tinha essa permissao isso passou; o `almoxarife_sesmt` foi o primeiro que nao
    tem, e para ele a primeira tela depois do login era um 403 — e a marca da
    lateral, que aponta para o comeco, era 403 em toda tela.

    Mover o painel de lugar resolveria tambem, e custaria mais: `/` e o caminho
    de sete telas de menu, de link no painel e de teste. O desvio deixa o painel
    onde esta e conserta o que estava errado, que era a porta da frente ser a
    tela de um modulo so.

    Nao exige permissao alem da sessao — como `/modulos`, e pelo mesmo motivo:
    quem chega aqui esta pedindo para ser levado a algum lugar, e negar seria
    negar a saida.
    """
    return RedirectResponse(modulos.primeira_tela(usuario), status_code=303)

File: nuvem/src/rotas/modulos.ts
/** /modulos — o mapa do sistema — e /inicio, o despachante da porta da frente. Porte de `app/rotas/modulos.py`. */
import { Hono } from "hono";
import type { Ambiente } from "../nucleo/contexto.js";
import { usuarioLogado } from "../dependencias.js";
import { pagina, redirecionar } from "../web.js";
import { primeira_tela } from "../modulos.js";

export const rotas = new Hono<Ambiente>();

rotas.get("/modulos", async (c) => {
  // sem exigir permissão: é a tela que explica onde ficam as coisas, e cada
  // item já se esconde sozinho de quem não pode vê-lo
  const usuario = await usuarioLogado(c);
  return pagina(c, "paginas/modulos.html", usuario);
});

/**
 * Onde o sistema começa para QUEM entrou — não é tela, é desvio. `/` é o
 * painel de Processos SEI e exige `processo.ver`; para o almoxarife a primeira
 * tela depois do login seria um 403. Não exige permissão além da sessão.
 */
rotas.get("/inicio", async (c) => {
  const usuario = await usuarioLogado(c);
  return redirecionar(c, primeira_tela(usuario));
});
